resize_data_part.target-sh

```diff
@@ -1,20 +1,47 @@
 #!/bin/sh
 
+set -e
+
 DEVICE=mmcblk0
 PART_NUM=2
 
 DEVICE_SIZE=`cat /sys/block/${DEVICE}/size`
 START=`partx /dev/${DEVICE} -n ${PART_NUM} -g -o start`
 SIZE=$((${DEVICE_SIZE} - ${START}))
 
-echo "${START},${SIZE}" | sfdisk --no-reread -uS -N ${PART_NUM} /dev/${DEVICE}
+mount_data_partition() {
+	echo "Mounting the partition..."
+	/bin/mount -o remount,rw /media
+	/bin/mount "/dev/${DEVICE}p${PART_NUM}" /media/data
+	/bin/mount -o remount,ro /media
+}
 
 clear
 echo 1 > /sys/devices/virtual/vtconsole/vtcon1/bind
 echo "Resizing the data partition. Please be patient."
 
-set +e
+# Unmount data partition before resizing for faster resizing.
+echo "Unmounting the partition..."
+/bin/mount -o remount,rw /media
+/bin/umount /media/data
+/bin/mount -o remount,ro /media
+# Try to remount the data partition on error (early EXIT is an error).
+trap mount_data_partition EXIT
+
+echo "Updating partition table..."
+
+# If the --no-reread option is not supported, this command will fail (|| true takes care of that)
```

Comment:
Why wouldn't `sfdisk` have a `--no-reread` option?

Reply:
sfdisk is part of util-linux which is tied to the kernel version
The util-linux for the older 3.12 kernel does not support the `--no-reread` option.

So it still tries to re-read. However, now it won't fail, because we've unmounted it first.

Reply:
The 3.12 does support no-reread. Looking into why it fails.

Reply:
`--no-reread` option is listed in `--help` but it still tries to re-read it:
![IMG_20200331_131725](https://user-images.githubusercontent.com/216339/78025540-22686180-7352-11ea-9180-f77cc87cb813.jpg)

Reply:
Ah, this is because `--no-reread` only affect the initial re-read. The final re-read after writing happens always (as per sfdisk 2.24.2 source)

Reply:
Turns returning non-zero on failure to re-read the table after write was a bug in older versions of `util-linux`. I've updated the RG350_buildroot util-linux to a slightly more recent version to fix the issue. https://github.com/tonyjih/RG350_buildroot/pull/67

Reply:
Fixed :)